src/views/HomeRedesign/HomeSection.tsx

```diff
@@ -0,0 +1,68 @@
+import { Heading } from "@chakra-ui/react";
+import { Fragment, FunctionComponent, useMemo } from "react";
+import { CatalogPackage } from "../../api/package/packages";
+import { PackageList } from "../../components/PackageList";
+import { useCatalog } from "../../contexts/Catalog";
+
+export interface HomeSectionProps {
+  name: string;
+  showLastUpdated?: number;
+  showPackages?: string[];
+}
+
+export const HomeSection: FunctionComponent<HomeSectionProps> = ({
+  name,
+  showLastUpdated,
+  showPackages,
+}) => {
+  const { data, loading, error } = useCatalog();
+
+  const results = useMemo(() => {
+    if (loading || error || !data?.packages) return [];
+
+    let packages = data.packages;
+
+    if (showLastUpdated) {
+      // sort packages by date
+      packages = [...packages]
+        .sort((p1, p2) => {
+          const d1 = new Date(p1.metadata.date);
+          const d2 = new Date(p2.metadata.date);
+          if (d1 === d2) {
+            return 0;
+          }
+          return d1 < d2 ? 1 : -1;
+        })
+        .slice(0, showLastUpdated);
+    } else if (showPackages) {
+      // find the specific packages from the catalog and
+      // return them in the same order
+      const filteredMap = [...packages].reduce(
+        (accum: { [key: string]: CatalogPackage }, p) => {
+          if (showPackages.includes(p.name)) {
+            return {
+              [p.name]: p,
+              ...accum,
+            };
+          } else {
+            return accum;
+          }
+        },
+        {}
+      );
+      packages = showPackages.map((p) => filteredMap[p]);
```

Comment:
There's an un-exported helper in `views/PackageLatest` which finds the latest version of a package by name. It may make sense to make that a generic util, and have something like the following:
```ts
packages = showPackages.map(findPackage);
```

Reply:
I wasn't sure what the best place to put the utility function was so I put it in `api/package/util.ts`.